sdk/src/AbstractMain.cpp

```diff
@@ -59,6 +59,42 @@ if (in_error)                             \
 
 namespace {
 
+int configureLoggingDir(
+   const system::FilePath& in_loggingDir,
+   const system::User& in_serverUser,
+   bool in_runUnprivilged)
+{ std::string message;
+
+   Error error = in_loggingDir.ensureDirectory();
+   CHECK_ERROR(error, "Invalid logging directory path - " + message)
+
+   // At this point the logging directory exists and is a directory. Make sure it belongs to the server user.
+   // First, check if the real user is root.
+   if (!in_runUnprivilged && system::posix::realUserIsRoot())
+   {
+      // If we are, restore root privileges.
+      error = system::posix::restoreRoot();
+      CHECK_ERROR(error, "Could not restore root privilege.")
```

Comment:
If it's possible to pull in the new `FileLogDestination` from upstream's `shared_core` we'd get all of this logic for free and could delete this entire method. Is that something we can do?

Reply:
I didn't pull the shared core in this branch, there is a seperate branch with those changes. I can use the newer shared_core FileLogDestination in this feature update, or update this method in the other branch.